libs/platform/src/lib/components/form/text-area/text-area.component.ts

```diff
@@ -0,0 +1,395 @@
+        } else {
+            // when custom value not set, we should set/reset counter to maxlength value when it becomes undefined
+            this.exceededCharCount = this.maxLength ? this.maxLength : 0;
+            // reset state by resetting value
+            super.writeValue('');
+        }
+    }
+
+    /** @hidden */
+    @ViewChild('textareaElement')
+    textareaElement: ElementRef;
+
+    /** @hidden */
+    hasTextExceeded: boolean = false;
+
+    /** @hidden excess character count */
+    exceededCharCount: number = this.maxLength ? this.maxLength : 0;
+
+    /** @hidden a string placeholder that toggles between 'remaining' and 'excess' for the select ICU expression */
+    counterExcessOrRemaining = 'remaining';
+
+    /** @hidden flag to check if there is an initial value set */
+    isValueCustomSet = false;
+
+    /** @hidden */
+    /** to track the current textarea content */
+    // textAreaValue: string;
+
+    /** @hidden */
+    /** to keep track of number of characters in the textarea */
+    private _textAreaCharCount: number = 0;
+
+    /** @hidden */
+    private _isPasted: boolean = false;
+
+    /** for i18n counter message translation */
+    private readonly remainingText = 'remaining';
+    private readonly excessText = 'excess';
+
+    constructor(
+        protected _cd: ChangeDetectorRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm
+    ) {
+        super(_cd, ngControl, ngForm);
+        if (this.ngControl) {
+            this.ngControl.valueAccessor = this;
+        }
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (!this.wrapType || VALID_WRAP_TYPES.indexOf(this.wrapType) === -1) {
+            throw new Error(`Textarea wrap type $ {this.wrapType} is not supported`);
+        }
+        // if not custom set, set counter to max length value, else it calculates remaining/exceeded characters.
+        if (!this.value) {
+            this.exceededCharCount = this.maxLength ? this.maxLength : 0;
+        } else {
+            this.isValueCustomSet = true;
+        }
+        super.ngOnInit();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        // set max height
+        this._setMaxHeight();
+
+        // don't set any error state if we are not showing counter message
+        if (!this.showExceededText) {
+            this.stateType = null;
+        }
+
+        super.ngAfterViewInit();
+    }
+
+    /** @hidden */
+    ngAfterViewChecked() {
+        // when value is custom set(initial value is present), the heights don't get
+        // set to show the full text in the ngAfterViewInit immediately. therefore, we call autoGrowTextArea in
+        // ngAfterViewChecked and detect changes again.
+        if (this.value && this.isValueCustomSet) {
+            this.autoGrowTextArea(); // if applicable, grow the textarea when value is custom set
+            this._cd.detectChanges();
```

Comment:
Why are you calling detect changes . Can you call simply stateChanges.emit() to inform the parent (formField). Maybe we dont need to call autoGrow everytime and only when it changed.

Reply:
calling stateChanges now.
autogrow method will only be called once `isValueCustomSet` is set. Once user edits the textarea or changes this value, `isValueCustomSet` is set to false and this block will not be run again.